### ISSUE TITLE
`kubeseal` adds trailing newline to YAML output

### ISSUE THREAD

**Issue by lindhe** [bitnami-labs/sealed-secrets#1296]
**Which component**:

`kubeseal`

**Describe the bug**

When creating a sealed secret (either a brand new one or amending to an existing one) with `kubeseal -o yaml`, the output has a trailing newline. This is considered bad practice by many, e.g. by Git (and maybe by the POSIX standard too, I'm not sure). Also, the corresponding JSON output is free of trailing newlines.

**To Reproduce**

```shell
kubectl create secret generic foo --dry-run=client -o yaml --from-literal=foo=bar | kubeseal -o yaml
```

**Expected behavior**

I expect there to be no trailing newline.

**Version of Kubernetes**:

n/a

**Additional context**

n/a

**Comment by lindhe** [comment]
I assume this is where it gets added:

https://github.com/bitnami-labs/sealed-secrets/blob/51eb3fb110204303ce4e73538e0e24a9485bc9d6/pkg/kubeseal/kubeseal.go#L353

But I also assume that it's there for a reason. Could we do something like `.trim()` to the object before printing it?

**Comment by lindhe** [comment]
This was resolved as of 0.24.0 (although #1330 was introduced instead).
